### PR TITLE
added carts for b2b

### PR DESCRIPTION
**Pull request by vadym-uskov**
- Added tests for /carts
- Added new "Save Header value to a variable:" method
- Updated the documentation for the created method

### CODE REVIEW

File: tests/api/b2b/glue/cart_endpoints/negative.robot [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    When I send a PATCH request:    /carts/${cart_id}    {"data": {"type": "carts","attributes": {"priceMode": "${net_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart124765"}}}
+    Then Response status code should be:    403
+    And Response reason should be:    Forbidden
+    And Response should return error message:    Missing access token.
+    And Response should return error code:    002
+    [Teardown]    Run Keywords    I set Headers:    Authorization=${token}
+    ...  AND    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Update_cart_with_non_existing_cart_id
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/cart_endpoints/negative.robot:234]: also need a test "with someone-else's cart id"

File: tests/api/b2b/glue/cart_endpoints/negative.robot [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+    When I send a DELETE request:    /carts/${cart_id}
+    Then Response status code should be:    403
+    And Response reason should be:    Forbidden
+    And Response should return error message:    Missing access token.
+    And Response should return error code:    002
+    [Teardown]    Run Keywords    I set Headers:    Authorization=${token}
+    ...  AND    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Delete_cart_with_invalid_cart_id
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/cart_endpoints/negative.robot:398]: also - someone-else's cart id

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 3]
```diff
@@ -0,0 +1,261 @@
+*** Settings ***
+Suite Setup    SuiteSetup
+Test Setup    TestSetup
+Resource    ../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+#GET requests
+Get_cart_by_cart_id
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot:9]: - added include=items
- other includes should be covered by other endpoints (added to the https://spryker.atlassian.net/wiki/spaces/PS/pages/3104834019/Robot+Framework+-+API+automation+progress table)

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+
+
+#POST requests
+Create_cart
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    When I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: - "test_cart_name" variable was added to the environments_api_b2b.json
- I added ${random} to the "test_cart_name" in tests

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    And Save value to a variable:    [data][id]    cart_id
+    And Save value to a variable:    [data][attributes][name]    cart_name
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    And Response body parameter should be:    [data][attributes][totals][priceToPay]    None
+    And Response body has correct self link for created entity:    ${cart_id}
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Create_cart_with_existing_name
+# Spryker is designed so that we can send existing name and it will be changed automatically to the unique value on the BE side.
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    ...  AND    Save value to a variable:    [data][id]    cart_id
+    ...  AND    Save value to a variable:    [data][attributes][name]    cart_name
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+
+
+#DELETE requests
+Delete_cart_by_cart_id
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+    And Response body parameter should be:    [data][attributes][totals][priceToPay]    0
+    And Response body has correct self link internal
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Update_cart_with_existing_name
+# Spryker is designed so that we can send existing name and it will be changed automatically to the unique value on the BE side.
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/negative.robot [hunk 10]
```diff
@@ -0,0 +1,423 @@
...
+    ...  AND    Response status code should be:    201
+    When I send a GET request:    /customers/user-01/carts
+    Then Response status code should be:    403
+    And Response reason should be:    Forbidden
+    And Response should return error code:    802
+    And Response should return error message:    Unauthorized request.
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Get_cart_without_customer_id
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/negative.robot:101]: added 'Get_cart_from_another_customer_id' test

File: tests/api/b2b/glue/cart_endpoints/negative.robot (hunk 1 above)
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/negative.robot:252]: added 'Update_cart_from_another_customer_cart_id' test

File: tests/api/b2b/glue/cart_endpoints/negative.robot (hunk 2 above)
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/negative.robot:435]: added 'Delete_cart_from_another_customer_id' test

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+    When I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
+    And Save value to a variable:    [data][id]    cart_id_2
+    Then Response status code should be:    201
+    And Response reason should be:    Created
+    And Response body parameter should be:    [data][type]    carts
+    And Response body parameter should be:    [data][id]    ${cart_id_2}
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    And Response body parameter should NOT be:    [data][attributes][name]    ${cart_name}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+    And Response body parameter should be:    [data][attributes][totals][priceToPay]    0
+    And Response body has correct self link internal
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Update_cart_with_empty_priceMod_currency_store
+# Spryker is designed so that we can send empty attributes: priceMod, currency, store and it will not be changed to the empty values.
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+    ...  AND    Save value to a variable:    [data][id]    cart_id
+    ...  AND    Save value to a variable:    [data][attributes][name]    cart_name
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+    ...  AND    I set Headers:    Authorization=${token}    If-Match=${header_tag}
+    When I send a PATCH request:    /carts/${cart_id}    {"data": {"type": "carts","attributes": {"priceMode": "","currency": "","store": ""}}}
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response body parameter should be:    [data][type]    carts
+    And Response body parameter should be:    [data][id]    ${cart_id}
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    And Response body parameter should be:    [data][attributes][name]    ${cart_name}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+    And Response body has correct self link for created entity:    ${cart_id_2}
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+    ...  AND    I send a DELETE request:    /carts/${cart_id_2}
+    ...  AND    Response status code should be:    204
+
+
+
+#PATCH requests
+Update_cart_by_cart_id
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: changed name to 'Update_cart_by_cart_id_with_all_attributes'

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 17]
```diff
@@ -0,0 +1,261 @@
...
+    And Response body parameter should be:    [data][attributes][totals][grandTotal]    0
+    And Response body parameter should be:    [data][attributes][totals][priceToPay]    0
+    And Response body has correct self link internal
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Update_cart_with_name_attribute
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "Test-cart"}}}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: fixed

File: tests/api/b2b/glue/cart_endpoints/negative.robot (hunk 10 above)
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/negative.robot:119]: Added 'Get_cart_by_cart_id_from_another_customer' test

File: tests/api/b2b/glue/cart_endpoints/positive.robot [hunk 18]
```diff
@@ -51,10 +50,65 @@ Get_cart_without_cart_id
...
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response body parameter should be:    [data][type]    carts
+    And Response body parameter should be:    [data][id]    df11d3f1-eb2f-508c-8a64-95c36b772712
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    And Response body parameter should be:    [data][attributes][name]    Newcomers
+    And Response body parameter should be:    [data][attributes][isDefault]    False
+    And Response body parameter should NOT be:    [data][attributes][totals][expenseTotal]    None
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: Changed keyword to the 'And Response body parameter should not be EMPTY:'

[the remark this answers is not in the document]
Comment (vadym-uskov) [reply tests/api/b2b/glue/cart_endpoints/positive.robot]: Added 'Should Not Be Empty ${data} ${data} is empty but should not be' line to the 'Response body parameter should NOT be:' keyword

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**